src/Avalonia.Controls/TextBox.cs

```diff
@@ -514,21 +514,36 @@ protected override void OnTextInput(TextInputEventArgs e)
 
         private void HandleTextInput(string input)
         {
-            if (!IsReadOnly)
+            if (IsReadOnly)
             {
-                input = RemoveInvalidCharacters(input);
-                string text = Text ?? string.Empty;
-                int caretIndex = CaretIndex;
-                if (!string.IsNullOrEmpty(input) && (MaxLength == 0 || input.Length + text.Length - (Math.Abs(SelectionStart - SelectionEnd)) <= MaxLength))
-                {
-                    DeleteSelection();
-                    caretIndex = CaretIndex;
-                    text = Text ?? string.Empty;
-                    SetTextInternal(text.Substring(0, caretIndex) + input + text.Substring(caretIndex));
-                    CaretIndex += input.Length;
-                    ClearSelection();
-                    _undoRedoHelper.DiscardRedo();
-                }
+                return;
+            }
+            
+            input = RemoveInvalidCharacters(input);
+            
+            if (string.IsNullOrEmpty(input))
+            {
+                return;
+            }
+            
+            string text = Text ?? string.Empty;
+            int caretIndex = CaretIndex;
+            int estimatedLength = input.Length + text.Length - Math.Abs(SelectionStart - SelectionEnd);
```

Comment:
Small question: why "estimated" length?

Reply:
Thanks @grokys! Its the calculated length before text input is actually being handled. This is the length "if" we don't trim text input.

Reply:
Ok, so it's not "estimated" as such. I'd propose `newLength` as the name for this. However because it's such a small nit, I'll make this change and merge if that's OK with your. Faster than the back-and-forth.